src/hardware/pounder/mod.rs

```diff
@@ -91,48 +93,87 @@ impl From<Channel> for GpioPin {
     }
 }
 
-#[derive(Serialize, Deserialize, Copy, Clone, Debug)]
-pub struct DdsChannelState {
-    pub phase_offset: f32,
+#[derive(Serialize, Deserialize, Copy, Clone, Debug, Miniconf)]
+pub struct DdsChannelConfig {
     pub frequency: f32,
+    pub phase_offset: f32,
     pub amplitude: f32,
-    pub enabled: bool,
 }
 
-#[derive(Serialize, Deserialize, Copy, Clone, Debug)]
-pub struct ChannelState {
-    pub parameters: DdsChannelState,
-    pub attenuation: f32,
+impl Default for DdsChannelConfig {
+    fn default() -> Self {
+        Self {
+            frequency: 10.0e6,
+            phase_offset: 0.0,
+            amplitude: 0.0,
+        }
+    }
 }
 
-#[derive(Serialize, Deserialize, Copy, Clone, Debug)]
-pub struct InputChannelState {
-    pub attenuation: f32,
-    pub power: f32,
-    pub mixer: DdsChannelState,
+impl DdsChannelConfig {
```

Comment:
Make a real `impl From<(ClockConfig, ChannelConfig)> for Profile` and do the conversion in that `impl`, not in `Profile::new()`.

Reply:
This would be implementing a trait for a struct in a foreign crate, `impl` it on a wrapper instead.